store/src/java/com/zimbra/cs/imap/RemoteImapMailboxStore.java

```diff
@@ -68,7 +68,27 @@ public RemoteImapMailboxStore(ZMailbox mailbox, String accountId) {
 
     @Override
     public ImapListener createListener(ImapFolder i4folder, ImapHandler handler) throws ServiceException {
-        return new ImapRemoteSession(this, i4folder, handler);
+        ImapRemoteSession irs = new ImapRemoteSession(this, i4folder, handler);
+        /* Without this, we may end up waiting for WaitSets to finish, expecting them to have
+         * spotted a change because the WaitSet last known change will be 0 but we want to be at least
+         * as up to date as the associated ZMailbox.  At this point, our view should be at least as
+         * up to date as ZMailbox, so this shouldn't cause loss of updates.
+         */
+        irs.updateLastChangeId(irs.getMailbox().getLastChangeID());
+        return irs;
+    }
+
+    @Override
+    public List<ImapListener> getListeners(ItemIdentifier ident) {
```

Comment:
The function 

    public List<ImapListener> getListeners(ItemIdentifier ident) {

is defined twice. Build should fails.
Please make sure you can build all components (native, common, soap, client and store) and run "ant test-all" before you send/update the PR.

Reply:
Fixed.

About this "Please run "git diff NICPS-155-check bugfix/NICPS-155". You can see a lot of changes in zimbra-attrs.xml" --

I saw changes, that are similar diff, same line is removed and added back.